### PR TITLE
Session and static

### PR DESCRIPTION
**Pull request by amir-arad**
add options for session and static serving of entire folders

### CODE REVIEW

File: scripts/src/middleware/session.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as session from "koa-session";
+import * as Koa from "koa";
+import Keygrip = require("keygrip");
+
+const DEFAULT_CONFIG: Config = {
+	// /**
```
Comment (nitzantomer) [inline scripts/src/middleware/session.ts]: all of the commented out code and comments are needed?
it looks like it's copied from somewhere, maybe just add the url as a comment above?

File: scripts/src/server.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ import { ExtendedArray } from "./common/array";
 
 import { middleware as statics } from "./middleware/statics";
 import { middleware as logging } from "./middleware/logging";
+import {Config as SessionConfig, middleware as session, Session} from "./middleware/session";
+import serve = require('koa-static');
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
double quotes: `require("koa-static")`

File: scripts/src/server.ts [hunk 3]
```diff
@@ -19,6 +19,8 @@ import { ExtendedArray } from "./common/array";
 
 import { middleware as statics } from "./middleware/statics";
 import { middleware as logging } from "./middleware/logging";
+import {Config as SessionConfig, middleware as session, Session} from "./middleware/session";
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
pad it with spaces:
`{ Config as SessionConfig, middleware as session, Session }`

File: scripts/src/server.ts [hunk 4]
```diff
@@ -35,6 +37,7 @@ export type RequestData = RequestDataGetter<string | any> & {
 	search: RequestDataGetter<string>;
 }
 export type Request = {
+	session: Session|undefined;
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add spaces:
`session: Session | undefined;`

File: scripts/src/server.ts [hunk 5]
```diff
@@ -145,15 +149,18 @@ export class ServerBuilder {
 	private _port: number;
 	private _host: string;
 	private _proxy: boolean;
+	private _session: Partial<SessionConfig>|undefined;
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add spaces:
`private _session: Partial<SessionConfig> | undefined;`

File: scripts/src/server.ts [hunk 6]
```diff
@@ -172,6 +179,10 @@ export class ServerBuilder {
 		return this;
 	}
 
+	session(config:Partial<SessionConfig> = {}){
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add spaces:
`session(config: Partial<SessionConfig> = {}) {`

File: scripts/src/server.ts [hunk 7]
```diff
@@ -316,18 +334,34 @@ class _Server implements Server {
 				cors: false | Cors.Options,
 				logger: winston.LoggerInstance,
 				files: Map<string, string>,
+				folders:Array<string>,
```
Comment (nitzantomer) [inline scripts/src/server.ts:337]: (formatting)
add space:
`folders: Array<string>,`

File: scripts/src/server.ts [hunk 8]
```diff
@@ -316,18 +334,34 @@ class _Server implements Server {
...
+				sessionConfig: Partial<SessionConfig>|undefined,
```
Comment (nitzantomer) [inline scripts/src/server.ts:338]: (formatting)
add spaces:
`sessionConfig: Partial<SessionConfig> | undefined,`

File: scripts/src/server.ts [hunk 9]
```diff
@@ -316,18 +334,34 @@ class _Server implements Server {
...
+			})
+		}
+
 		this.setupLogging();
 
 		if (cors) {
 			this.koa.use(Cors(cors));
 		}
 
+		if(sessionConfig !== undefined) {
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add space:
`if (sessionConfig !== undefined) {`

File: scripts/src/server.ts [hunk 10]
```diff
@@ -316,18 +334,34 @@ class _Server implements Server {
...
+			if(sessionConfig.keygrip) {
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add space:
`if (sessionConfig.keygrip) {`

File: scripts/src/server.ts [hunk 11]
```diff
@@ -316,18 +334,34 @@ class _Server implements Server {
...
+				sessionConfig: Partial<SessionConfig>|undefined,
 				routes: Route[]) {
 		this.host = host;
 		this.port = port;
 		this.logger = logger;
 
 		this.koa = new Koa();
+
+		if (folders.length) {
+			folders.forEach((folder)=>{
```
Comment (nitzantomer) [inline scripts/src/server.ts]: (formatting)
add spaces, also no need to wrap one param in brackets:
`folders.forEach(folder => {`

**Review by nitzantomer — CHANGES_REQUESTED** [review]

File: scripts/src/server.ts (hunk 11 above)
Comment (amir-arad) [inline scripts/src/server.ts]: unsure where to add spaces

Comment (amir-arad) [reply scripts/src/server.ts]: unsure where to add spaces

**Pull request merged**